File: frontend/src/app/common/restService/TicketService.ts
/**
 * Created by zhaowei on 2017/10/16.
 */
import {Injectable} from '@angular/core';
import {Http} from '@angular/http';
import HandleError from './../service/HandleError';
import {Ticket} from '../class/Ticket';
@Injectable()
export class TicketService {

  constructor(private http: Http) {
  }

  /**获取用户列表**/
  private url = '/cfmy/ticket/';

  // 添加
  public add(ticket: Ticket): Promise<void> {
    return this.http.post(this.url + 'add',
      JSON.stringify(ticket))
      .toPromise()
      .then(response => response.json())
      .catch(HandleError);
  }

  public list(obj: Object, body: Object): Promise<void> {
    return this.http.post(this.url + 'list/' + (obj as any).pageNum + '/' + (obj as any).pageSize,
      JSON.stringify(body ? body : {}))
      .toPromise()
      .then(response => response.json())
      .catch(HandleError);
  }

  public getById(id: String): Promise<void> {
    return this.http.get(this.url + 'getById?id=' + id)
      .toPromise()
      .then(response => response.json())
      .catch(HandleError);
  }

  public update(ticket: Ticket): Promise<void> {
    return this.http.post(this.url + 'update',
      JSON.stringify(ticket))
      .toPromise()
      .then(response => response.json())
      .catch(HandleError);
  }

}

File: src/main/java/com/zw/cf/service/TicketService.java
package com.zw.cf.service;

import com.zw.cf.model.Ticket;
import com.zw.cf.vo.TicketListFind;
import com.zw.plug.Response;

/**
 * Created by zhaowei on 2017/8/17.
 */
public interface TicketService {
    Response add(Ticket ticket);
    Response list(Integer pageNum, Integer pageSize, TicketListFind ticketListFind);
    Response getById(String id);
    Response update(Ticket ticket);
    Response updateState(String id,Integer state);
    Response del(String id);
}
